defender/staged/downloaded/Compound_Gasless_Voting.js

```diff
@@ -0,0 +1,95 @@
+const ethers = require('ethers');
+
+const { DefenderRelayProvider, DefenderRelaySigner } = require('defender-relay-client/lib/ethers');
+
+const compoundGovernanceAddress = '0xc0Da02939E1441F497fd74F78cE7Decb17B66529'; // GovernorBravoDelegate.sol
+
+const compTokenAddress = '0xc00e94Cb662C3520282E6f5717214004A7f26888'; // Comp.sol
+
+const compoundGovernanceAbi = [
+  'function castVoteBySig(uint proposalId, uint8 support, uint8 v, bytes32 r, bytes32 s)',
+];
+
+const compTokenAbi = [
+  'function delegateBySig(address delegatee, uint nonce, uint expiry, uint8 v, bytes32 r, bytes32 s)',
+];
+
+exports.handler = async function handler(autotaskEvent) {
+  if (autotaskEvent.request === undefined) {
+    throw new Error('POST request information not found');
+  }
+  // body contains batch of signed messages to submit on-chain through the Defender Relayer
+  // the api endpoint does most of the validation checking for us already
+  // (ex. checks for valid signatures and all required fields)
+  // https://app.swaggerhub.com/apis-docs/arr00/COMP.vote/1.0#/
+  const { body } = autotaskEvent.request;
+
+  if (Array.isArray(body) === false) {
+    throw new Error('Request body must be an Array');
+  }
+
+  // initiate a new relay provider
+  console.debug('Creating DefenderRelayProvider');
+  const provider = new DefenderRelayProvider(autotaskEvent);
+
+  // initiate a new relay signer
+  console.debug('Creating DefenderRelaySigner');
+  const signer = new DefenderRelaySigner(autotaskEvent, provider, { speed: 'fast' });
+
+  // create an ethers.js Contract Object to interact with the on-chain governance smart contract
+  console.debug('Creating governanceContract');
+  const governanceContract = new ethers.Contract(
+    compoundGovernanceAddress,
+    compoundGovernanceAbi,
+    signer,
+  );
+
+  // create an ethers.js Contract Object to interact with the on-chain COMP smart contract
+  console.debug('Creating compContract');
+  const compTokenContract = new ethers.Contract(
+    compTokenAddress,
+    compTokenAbi,
+    signer,
+  );
+
+  // cast vote or delegate vote on-chain
+  const promises = body.map(async (votes) => {
+    if (votes.delegatee !== undefined) {
+      const {
+        address, delegatee, nonce, expiry, v, r, s,
+      } = votes;
+
+      // delegate vote on-chain
+      console.debug(`Delgating votes from ${address} to ${delegatee}`);
+      try {
+        await compTokenContract.delegateBySig(delegatee, nonce, expiry, v, r, s);
+      } catch (error) {
+        console.error(error);
+      }
+    } else if (votes.support !== undefined) {
+      const {
+        address, proposalId, support, v, r, s,
```

Comment:
Should we sanity check these values before spending the gas to make the call?

Reply:
added sanity check to make sure address had delegated votes and proposal Id was active